src/ssl.c

```diff
@@ -34723,7 +34723,7 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
             int    length;
             word32 idx = 0;
 
-            if (a->obj[idx++] != ASN_OBJECT_ID) {
+            if (CheckASNTag(a->obj, ASN_OBJECT_ID, idx++, a->objSz) != 0) {
```

Comment:
This doesn't follow the pattern of other asn functions.

1) The index is incremented before verifying there is room to do so
2) The index is not passed by pointer so that the calling function can correctly set the index if in fact it is adjusted (after checking)

Reply:
After the review am in the process of changing the function to be:

```
-WOLFSSL_LOCAL int CheckASNTag(const byte* input, byte tag, word32 idx,
-                              word32 inputSz);
+WOLFSSL_LOCAL int GetASNTag(const byte* input, word32* idx, byte* tag,
+                            word32 inputSz);
```